www/__tests__/enketoHelper.test.ts

```diff
@@ -0,0 +1,245 @@
+  const opts = { prefilledSurveyResponse: filled };
+  const opts2 = {
+    prefillFields: {
+      Start_date: '2016-07-25',
+      Start_time: '17:24:32.928-06:00',
+      End_date: '2016-07-25',
+      End_time: '17:30:31.000-06:00',
+    },
+  };
+
+  //if no xmlModel, returns null
+  expect(getInstanceStr(null, opts)).toBe(null);
+  //if there is a prefilled survey, return it
+  expect(getInstanceStr(xmlModel, opts)).toBe(filled);
+  //if there is a model and fields, return prefilled
+  expect(getInstanceStr(xmlModel, opts2)).toBe(filled);
+  //if none of those things, also return null
+  expect(getInstanceStr(xmlModel, {})).toBe(null);
+});
+
+//resolve timestamps
+it('resolves the timestamps', () => {
+  const xmlParser = new window.DOMParser();
+  const timelineEntry = {
+    end_local_dt: { timezone: 'America/Los_Angeles' },
+    start_ts: 1469492672.928242,
+    end_ts: 1469493031,
+  };
+
+  //missing data returns null
+  const missingData =
+    '<tag> <Start_date>2016-08-28</Start_date> <End_date>2016-07-25</End_date> <End_time>17:30:31.000-06:00</End_time> </tag>';
+  const missDataDoc = xmlParser.parseFromString(missingData, 'text/html');
+  expect(resolveTimestamps(missDataDoc, timelineEntry)).toBeNull();
+  //bad time returns undefined
+  const badTimes =
+    '<tag> <Start_date>2016-08-28</Start_date> <End_date>2016-07-25</End_date> <Start_time>17:32:32.928-06:00</Start_time> <End_time>17:30:31.000-06:00</End_time> </tag>';
+  const badTimeDoc = xmlParser.parseFromString(badTimes, 'text/xml');
+  expect(resolveTimestamps(badTimeDoc, timelineEntry)).toBeUndefined();
+  //good info returns unix start and end timestamps -- TODO : address precise vs less precise?
+  const timeSurvey =
+    '<tag> <Start_date>2016-07-25</Start_date> <End_date>2016-07-25</End_date> <Start_time>17:24:32.928-06:00</Start_time> <End_time>17:30:31.000-06:00</End_time> </tag>';
+  const xmlDoc = xmlParser.parseFromString(timeSurvey, 'text/xml');
+  expect(resolveTimestamps(xmlDoc, timelineEntry)).toMatchObject({
+    start_ts: 1469492672.928242,
+    end_ts: 1469493031,
+  });
+});
+
+//resolve label
+it('resolves the label', async () => {
+  const xmlParser = new window.DOMParser();
+  const xmlString = '<tag> <Domestic_activities> option_1 </Domestic_activities> </tag>';
+  const xmlDoc = xmlParser.parseFromString(xmlString, 'text/html');
+  const xmlString2 =
+    '<tag> <Domestic_activities> option_1 </Domestic_activities> <Employment_related_a_Education_activities> option_3 </Employment_related_a_Education_activities> </tag>';
+  const xmlDoc2 = xmlParser.parseFromString(xmlString2, 'text/xml');
+
+  //if no template, returns "Answered" TODO: find a way to engineer this case
```

Comment:
To engineer this case you would need a different mock config

Reply:
implemented the parameter in the mock, added test cases in [80aa5ea](https://github.com/e-mission/e-mission-phone/pull/1063/commits/80aa5ea18081849819236d65dc5d07ea583aa8e8)